lib/sigh/developer_center.rb

```diff
@@ -197,37 +192,27 @@ def maintain_app_certificate(app_identifier, type, force, cert_date)
         certs = post_ajax(@list_certs_url)
 
         Helper.log.info "Checking if profile is available. (#{certs['provisioningProfiles'].count} profiles found)"
+        required_cert_types = type == DEVELOPMENT ? ['iOS Development'] : ['iOS Distribution', 'iOS UniversalDistribution']
         certs['provisioningProfiles'].each do |current_cert|
-          if type == DEVELOPMENT
-            if current_cert['type'] != "iOS Development"
-              next
-            end
-          end
-
-          if type != DEVELOPMENT
-            if not ['iOS Distribution', 'iOS UniversalDistribution'].include?current_cert['type']
-              next
-            end
-          end
+          next unless required_cert_types.include?(current_cert['type'])
           
           details = profile_details(current_cert['provisioningProfileId'])
 
           if details['provisioningProfile']['appId']['identifier'] == app_identifier
-            if type == APPSTORE and details['provisioningProfile']['deviceCount'] > 0
-              next # that's an Ad Hoc profile. I didn't find a better way to detect if it's one ... skipping it
-            end
-            if type != APPSTORE and details['provisioningProfile']['deviceCount'] == 0
-              next # that's an App Store profile ... skipping it
-            end
+            # that's an Ad Hoc profile. I didn't find a better way to detect if it's one ... skipping it
+            next if type == APPSTORE && details['provisioningProfile']['deviceCount'] > 0
+
+            # that's an App Store profile ... skipping it
+            next if type != APPSTORE && details['provisioningProfile']['deviceCount'] == 0
```

Comment:
I can still forcefully renew an enterprise profile. So no problems from this side.